src/libraries/xsync/src/xsync_store.cpp

```diff
@@ -170,7 +170,13 @@ uint64_t xsync_store_t::get_latest_end_block_height(const std::string & account,
             return 1;
         }
     }
-    return connect_height + 2;
+
+    if (sync_policy == enum_chain_sync_policy_checkpoint) {
+        return connect_height;
```

Comment:
if mutable connectted height == immutable height, end height is cp height;  else  end height is cp+2;